ntbs-service/DataMigration/NotificationImportService.cs

```diff
@@ -0,0 +1,126 @@
+        {
+            LogInformation($"Request to import Notification with Id={notificationId}");
+
+            if (notificationRepository.NotificationWithLtbrOrEtsIdExists(notificationId)) 
+            {
+                LogInformation($"Notification with Id={notificationId} already exists in database");
+                return null;
+            }
+
+            var notificationTask = notificationSearcher.Search(notificationId);
+            Task.WaitAll(notificationTask);
+            var notifications = notificationTask.Result.ToList();
+
+            LogInformation($"{notifications.Count} notifications included linked ones found to import");
+            var notificationsToSave = new List<Notification>();
+            foreach (var notification in notifications)
+            {
+                var linkedNotificationId = notification.LTBRID ?? notification.ETSID;  
+                LogInformation($"Validating notification with Id={linkedNotificationId}");
+
+                var validationErrors = GetValidationErrors(notification);
+                
+                if (validationErrors.Count() == 0)
+                {
+                    notificationsToSave.Add(notification);
+                    LogInformation($"No validation errors found");
+                }
+                else
+                {
+                    LogInformation($"{validationErrors.Count()} validation errors found:");
+                    foreach (var validationError in validationErrors)
+                    {
+                        LogInformation(validationError.ErrorMessage);
+                    }
+                }
+            }
+            LogInformation($"Importing {notificationsToSave.Count()} valid notifications");
+            var savedNotificaitons = await AddLinkedNotificationsAsync(notificationsToSave);
+
+
+            LogInformation($"Finished importing notification with Id={notificationId} and linked notifications");
+            return savedNotificaitons;
+        }
+
+        private IEnumerable<ValidationResult> GetValidationErrors(Notification notification) 
+        {
+            var validationsResults = new List<ValidationResult>();
+
+            notification.ShouldValidateFull = true;  
+            notification.PatientDetails.ShouldValidateFull = true;  
+            notification.ClinicalDetails.ShouldValidateFull = true;  
+            notification.TravelDetails.ShouldValidateFull = true;  
+            notification.VisitorDetails.ShouldValidateFull = true;  
+
+            validationsResults.AddRange(ValidateObject(notification));
+            validationsResults.AddRange(ValidateObject(notification.PatientDetails));
+            validationsResults.AddRange(ValidateObject(notification.ClinicalDetails));
+            validationsResults.AddRange(ValidateObject(notification.TravelDetails));
+            validationsResults.AddRange(ValidateObject(notification.VisitorDetails));

```

Comment:
Hmm, my experience with **Model** validation methods `TryValidateModel` on page controllers suggested that they DO validate recursively.. Bit of a shame that it's not consistent - oh well!

Perhaps it's worth ensuring that we pick all sub-models here, not just the ones we're currently supplying data for?

Also, as the `TryValidateObject` method already asks for it, should we supply it, rather than re-creating lists and adding their contests to the master list? So something like:
```
new List<object> { notification, notification.PatientDetails, notification.ClinicalDetails, ...}
  .ForEach(objectToValidate => {
     var validationContext = new ValidationContext(objectToValidate);
     Validator.TryValidateObject(objectToValidate, validationContext , validationsResults, true);
  }
```